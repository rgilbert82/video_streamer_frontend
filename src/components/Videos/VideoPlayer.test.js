import React           from 'react';
import { shallow }     from 'enzyme';
import { VideoPlayer } from '.';

describe(VideoPlayer, () => {
  const video = {
    id:           '123',
    title:        'Title',
    description:  'Description',
    published_at: '2018'
  };
  let component;

  beforeEach(() => {
    component = shallow(<VideoPlayer video={video} />);
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });
});
