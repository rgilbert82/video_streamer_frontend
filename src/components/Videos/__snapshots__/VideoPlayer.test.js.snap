// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class VideoPlayer extends _react2.default.Component {
  render() {
    const videoUrl = \`https://www.youtube.com/embed/\${this.props.video.id}?autoplay=1\`;
    return (
      _react2.default.createElement("div", { __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: this },
        _react2.default.createElement("h2", { __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this }, this.props.video.title),
        _react2.default.createElement("iframe", { width: "560", height: "315", src: videoUrl, frameBorder: "0", allow: "autoplay; encrypted-media", allowFullScreen: true, __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this }),
        _react2.default.createElement("p", { __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this }, this.props.video.description),
        _react2.default.createElement("small", { __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this }, this.props.video.published_at)));


  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    video={
        Object {
            "description": "Description",
            "id": "123",
            "published_at": "2018",
            "title": "Title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Title
</h2>,
        <iframe
          allow="autoplay; encrypted-media"
          allowFullScreen={true}
          frameBorder="0"
          height="315"
          src="https://www.youtube.com/embed/123?autoplay=1"
          width="560"
/>,
        <p>
          Description
</p>,
        <small>
          2018
</small>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "allow": "autoplay; encrypted-media",
          "allowFullScreen": true,
          "frameBorder": "0",
          "height": "315",
          "src": "https://www.youtube.com/embed/123?autoplay=1",
          "width": "560",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Description",
        },
        "ref": null,
        "rendered": "Description",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018",
        },
        "ref": null,
        "rendered": "2018",
        "type": "small",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Title
</h2>,
          <iframe
            allow="autoplay; encrypted-media"
            allowFullScreen={true}
            frameBorder="0"
            height="315"
            src="https://www.youtube.com/embed/123?autoplay=1"
            width="560"
/>,
          <p>
            Description
</p>,
          <small>
            2018
</small>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "allow": "autoplay; encrypted-media",
            "allowFullScreen": true,
            "frameBorder": "0",
            "height": "315",
            "src": "https://www.youtube.com/embed/123?autoplay=1",
            "width": "560",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Description",
          },
          "ref": null,
          "rendered": "Description",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018",
          },
          "ref": null,
          "rendered": "2018",
          "type": "small",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`class VideoPlayer extends _react2.default.Component {
  render() {
    const videoUrl = \`https://www.youtube.com/embed/\${this.props.video.id}?autoplay=1\`;
    return (
      _react2.default.createElement("div", { __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: this },
        _react2.default.createElement("h2", { __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this }, this.props.video.title),
        _react2.default.createElement("iframe", { width: "560", height: "315", src: videoUrl, title: props.video.id, frameBorder: "0", allow: "autoplay; encrypted-media", allowFullScreen: true, __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this }),
        _react2.default.createElement("p", { __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this }, this.props.video.description),
        _react2.default.createElement("small", { __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this }, this.props.video.published_at)));


  }} renders properly 1`] = `undefined`;

exports[`class VideoPlayer extends _react2.default.Component {
  render() {
    const videoUrl = \`https://www.youtube.com/embed/\${this.props.video.id}?autoplay=1\`;
    return (
      _react2.default.createElement("div", { __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: this },
        _react2.default.createElement("h2", { __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this }, this.props.video.title),
        _react2.default.createElement("iframe", { width: "560", height: "315", src: videoUrl, title: this.props.video.id, frameBorder: "0", allow: "autoplay; encrypted-media", allowFullScreen: true, __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this }),
        _react2.default.createElement("p", { __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this }, this.props.video.description),
        _react2.default.createElement("small", { __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this }, this.props.video.published_at)));


  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    video={
        Object {
            "description": "Description",
            "id": "123",
            "published_at": "2018",
            "title": "Title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Title
</h2>,
        <iframe
          allow="autoplay; encrypted-media"
          allowFullScreen={true}
          frameBorder="0"
          height="315"
          src="https://www.youtube.com/embed/123?autoplay=1"
          title="123"
          width="560"
/>,
        <p>
          Description
</p>,
        <small>
          2018
</small>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "allow": "autoplay; encrypted-media",
          "allowFullScreen": true,
          "frameBorder": "0",
          "height": "315",
          "src": "https://www.youtube.com/embed/123?autoplay=1",
          "title": "123",
          "width": "560",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Description",
        },
        "ref": null,
        "rendered": "Description",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018",
        },
        "ref": null,
        "rendered": "2018",
        "type": "small",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Title
</h2>,
          <iframe
            allow="autoplay; encrypted-media"
            allowFullScreen={true}
            frameBorder="0"
            height="315"
            src="https://www.youtube.com/embed/123?autoplay=1"
            title="123"
            width="560"
/>,
          <p>
            Description
</p>,
          <small>
            2018
</small>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "allow": "autoplay; encrypted-media",
            "allowFullScreen": true,
            "frameBorder": "0",
            "height": "315",
            "src": "https://www.youtube.com/embed/123?autoplay=1",
            "title": "123",
            "width": "560",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Description",
          },
          "ref": null,
          "rendered": "Description",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018",
          },
          "ref": null,
          "rendered": "2018",
          "type": "small",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
